File: ASSIGNMENT9/c/Q.9.c
#include <stdio.h>

int findMinIndexRepeating(int arr[], int n) {
    int minIndex = n;
    int index[100] = {0};

    for (int i = 0; i < n; i++) {
        if (index[arr[i]] != 0) {
            if (index[arr[i]] - 1 < minIndex) {
                minIndex = index[arr[i]] - 1;
            }
        } else {
            index[arr[i]] = i + 1;
        }
    }

    return (minIndex == n) ? -1 : minIndex;
}

int main() {
    int arr[] = {4, 5, 6, 4, 5, 3};
    int n = sizeof(arr) / sizeof(arr[0]);
    int minIndex = findMinIndexRepeating(arr, n);
    if (minIndex != -1) {
        printf("Minimum index of repeating element is %d\n", minIndex);
    } else {
        printf("No repeating elements found\n");
    }
    return 0;
}

File: ASSIGNMENT9/python/Q.9.py
def min_index_of_repeating_element(arr):
    seen = {}
    min_index = len(arr)
    for i, num in enumerate(arr):
        if num in seen:
            min_index = min(min_index, seen[num])
        else:
            seen[num] = i
    return min_index if min_index != len(arr) else None

arr = [1, 2, 3, 2, 1]
print("Minimum index of repeating element:", min_index_of_repeating_element(arr))
